src/clock.rs

```diff
@@ -81,3 +81,69 @@ impl embedded_time::Clock for TestClock {
         Ok(embedded_time::Instant::new(0))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use embedded_time::duration::{Milliseconds, Seconds};
+
+    #[test]
+    fn test_checked_add() {
+        let clock = TestClock::new();
+        let now = clock.try_now().unwrap();
+        assert_eq!(
+            now.checked_add(Seconds::new(1u64)),
+            Some(CtapInstant::new(TEST_CLOCK_FREQUENCY_HZ as u64))
+        );
+        assert_eq!(
+            now.checked_add(Seconds::new(1u64)),
+            now.checked_add(Milliseconds::new(1000u64))
+        );
+    }
+
+    #[test]
+    fn test_checked_add_overflow() {
+        assert_eq!(
+            CtapInstant::new(u64::MAX).checked_add(Seconds::new(1u64)),
+            Some(CtapInstant::new(TEST_CLOCK_FREQUENCY_HZ as u64 - 1u64))
+        );
+    }
+
+    #[test]
+    fn test_checked_add_error() {
+        assert!(CtapInstant::new(u64::MAX)
```

Comment:
This doesn't need to be MAX for the test and I think it's less confusing to use 0 (or now like you do below) instead, to make it obvious the problem is not an overflow, it's just the duration which is too big.